tests/foreman/virtwho/test_api_virtwho.py

```diff
@@ -107,3 +111,243 @@ def test_positive_deploy_configure_id(self):
+        name = gen_string('alpha')
+        args = self._make_virtwho_configure()
+        args.update({'name': name})
+        vhd = entities.VirtWhoConfig(**args).create()
+        options = {
+            '60': '3600',
+            '120': '7200',
+            '240': '14400',
+            '480': '28800',
+            '720': '43200',
+            '1440': '86400',
+            '2880': '172800',
+            '4320': '259200',
+        }
+        for key, value in sorted(
+                options.items(),
+                key=lambda item: int(item[0])):
+            vhd.interval = key
+            vhd.update(['interval'])
+            command = get_configure_command(vhd.id)
+            deploy_configure_by_command(command)
+            self.assertEqual(
+                get_configure_option('VIRTWHO_INTERVAL', VIRTWHO_SYSCONFIG),
+                value)
+        vhd.delete()
+        self.assertEqual(entities.VirtWhoConfig().search(
+            query={'search': 'name={}'.format(name)}), [])
+
+    @tier1
+    def test_positive_hypervisor_id_option(self):
+        """ Verify hypervisor_id option by "PUT
+
+        /foreman_virt_who_configure/api/v2/configs/:id"
+
+        :id: 9aa17bbc-e417-473a-831c-4d87781f41d8
+
+        :expectedresults: hypervisor_id option can be updated.
+
+        :CaseLevel: Integration
+
+        :CaseImportance: Medium
+        """
+        name = gen_string('alpha')
+        args = self._make_virtwho_configure()
+        args.update({'name': name})
+        vhd = entities.VirtWhoConfig(**args).create()
+        values = ['uuid', 'hostname']
+        if self.hypervisor_type in ('esx', 'rhevm'):
+            values.append('hwuuid')
+        for value in values:
+            vhd.hypervisor_id = value
+            vhd.update(['hypervisor_id'])
+            config_file = get_configure_file(vhd.id)
+            command = get_configure_command(vhd.id)
+            deploy_configure_by_command(command)
+            self.assertEqual(
+                get_configure_option('hypervisor_id', config_file),
+                value
+            )
+        vhd.delete()
+        self.assertEqual(entities.VirtWhoConfig().search(
+            query={'search': 'name={}'.format(name)}), [])
+
+    @tier1
+    def test_positive_filter_option(self):
+        """ Verify filter option by "PUT
+
+        /foreman_virt_who_configure/api/v2/configs/:id"
+
+        :id: 89cc1134-69d9-4da8-9ba9-a296c17f4f16
+
+        :expectedresults: filter and filter_hosts can be updated.
+
+        :CaseLevel: Integration
+
+        :CaseImportance: Medium
+        """
+        name = gen_string('alpha')
+        args = self._make_virtwho_configure()
+        args.update({'name': name})
+        vhd = entities.VirtWhoConfig(**args).create()
+        whitelist = {
+            'filtering_mode': '1',
+            'whitelist': '.*redhat.com'
+        }
+        blacklist = {
+            'filtering_mode': '2',
+            'blacklist': '.*redhat.com'
+        }
```

Comment:
How about the `filter_host_parents` and `exclude_host_parents`? is it available for API?

Reply:
nope, can not set it via API/CLI as bug: https://bugzilla.redhat.com/show_bug.cgi?id=1732699